test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs

```diff
@@ -0,0 +1,577 @@
+//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Relay.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net;
+    using System.Net.WebSockets;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class RunTimeTests : HybridConnectionTestBase
+    {
+        public RunTimeTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task UnauthenticatedHybridConnection()
+        {
+            this.Logger.Log("Creating a listener connection string for the unauthenticated Hybrid Connection");
+            var listenerConnectionStringBuilder = this.ConnectionStringBuilder;
+            listenerConnectionStringBuilder.EntityPath = "unauthenticated";
+            var listenerConnectionString = listenerConnectionStringBuilder.ToString();
+
+            this.Logger.Log("Creating a client connection string for the unauthenticated Hybrid Connection. Setting the keys to string.Empty");
+            var clientConnectionStringBuilder = this.ConnectionStringBuilder;
```

Comment:
Especially since it clears out the SAS values that the connection string gave us.